frontend/common/kubeflow-common-lib/projects/kubeflow/src/lib/status-icon/status-icon.component.ts

```diff
@@ -7,26 +7,27 @@ import { STATUS_TYPE } from '../resource-table/status/types';
   styleUrls: ['./status-icon.component.scss'],
 })
 export class StatusIconComponent {
-  @Input() status: STATUS_TYPE;
-  @Input() stateChanging: boolean;
+  @Input() phase: STATUS_TYPE;
+
+  STATUS_TYPE = STATUS_TYPE;
 
   get statusIcon(): string {
-    if (this.status === STATUS_TYPE.WARNING) {
-      return 'warning';
-    }
-    if (
-      this.status === STATUS_TYPE.WAITING ||
-      this.status === STATUS_TYPE.TERMINATING
-    ) {
-      return 'timelapse';
-    } else if (this.status === STATUS_TYPE.READY) {
-      return 'check_circle';
-    } else if (this.status === STATUS_TYPE.STOPPED) {
-      return 'stop_circle';
-    } else {
-      return STATUS_TYPE.UNINITIALIZED;
+    switch (this.phase) {
+      case STATUS_TYPE.READY: {
+        return 'check_circle';
+      }
+      case STATUS_TYPE.WARNING: {
+        return 'warning';
+      }
+      case STATUS_TYPE.UNAVAILABLE: {
+        return 'timelapse';
+      }
+      case STATUS_TYPE.ERROR: {
+        return 'error';
+      }
+      default: {
+        return 'warning';
+      }
```

Comment:
Why have an unavailable class if not in switch case
Or why have Unavailable status if return timelapse

Reply:
Not sure I understand.
This is just taking changes from upstream. Thhere is an "unavailable" case check in the switch. And it just returns an icon type, probably no such thing as an "unavailable" icon, so it uses the timelapse icon.